src/components/Button/Button.tsx

```diff
@@ -2,33 +2,77 @@ import React from "react";
 import styled, { css } from "styled-components";
 
 export interface ButtonProps {
-  primary?: boolean;
   children: string;
+  variant: "filled" | "outlined";
+  theme: "light" | "dark" | "blue";
+  disabled: boolean;
   onClick?: () => void;
 }
 
 /**
  * Primary UI component for user interaction
  */
 export const Button = ({ children, ...props }: ButtonProps) => (
-  <StyledButton type="button" {...props}>
-    {children}
-  </StyledButton>
+  <StyledButton {...props}>{children}</StyledButton>
 );
 
+const FILLED = ({ theme }: ButtonProps) => css`
+  color: #ffffff;
+  background-color: #0657f9;
+  font-weight: 700;
+
+  &:hover {
+    background-color: #004adf;
+  }
+
+  &:disabled {
+    color: #ced5e3;
+    background-color: #f4f7fb;
+    cursor: not-allowed;
+  }
+`;
+
+const OUTLINED = ({ theme }) => css`
+  color: #0657f9;
+  background-color: transparent;
+  font-weight: 400;
+  border: 1px solid #0657f9;
+  box-sizing: border-box;
+
+  &:hover {
+    background-color: #e0eaff;
+  }
+
+  &:disabled {
+    color: #ced5e3;
+    background-color: transparent;
+    border: 1px solid #ced5e3;
+    cursor: not-allowed;
+  }
+`;
+
+const VARIANTS = {
+  filled: FILLED,
+  outlined: OUTLINED,
+};
+
 const StyledButton = styled.button`
-  font-family: "Roboto";
-  font-weight: "bold";
-  border: 0;
-  border-radius: 3em;
-  padding: 8px 24px;
-  cursor: pointer;
-  display: inline-block;
-  line-height: 1;
-  color: white;
-  ${({ primary }: ButtonProps) =>
-    primary &&
-    css`
-      background-color: #0657f9;
-    `}
+  ${({ variant }: ButtonProps) => css`
+    width: 220px;
+    height: 40px;
+    font-family: "Roboto";
```

Comment:
Maybe we should add some fallbacks here, just to be sure

Reply:
Do we have a preferred fallback within yearn apps?

Reply:
We don't. I think you could use something standard/safe like that
```
font-family: "Roboto", ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", Arial, "Noto Sans", sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
```

Reply:
👍 Added with https://github.com/Major-Eth/yearn-web-lib/pull/9/commits/16da1b2011219eaf49c78ebe645b29642aa3327c